src/LogicTasks/Semantics/Decide.hs

```diff
@@ -107,44 +107,70 @@ verifyQuiz DecideConfig{..}
 start :: [Int]
 start = []
 
-
+preventIfSolutionExceedsTableSize :: OutputMonad m => Int -> Table -> LangM m
+preventIfSolutionExceedsTableSize solLen table = preventWithHint (solLen > tableLen)
+    (translate $ do
+      german "Lösung überschreitet nicht Anzahl der Zeilen?"
+      english "Solution does not exceed count of rows?"
+    )
+    (translate $ do
+      german $ "Ihre Lösung enthält mehr Werte als es Zeilen gibt. " ++ gerHint
+      english $ "Your solution contains more values than there are rows. " ++ engHint
+    )
+  where
+    tableLen = length $ readEntries table
+    diffToTable = abs (solLen - tableLen)
+    (gerLong,engLong) = gerEng diffToTable
+    (gerHint,engHint) = (
+      "Es " ++ gerLong ++" entfernt werden.",
+      "Please remove at least " ++ engLong ++ " to proceed."
+      )
+    gerEng diff = if diff == 1
+        then ("muss mindestens " ++ display ++ " Wert", display ++ " value") -- no-spell-check
+        else ("müssen mindestens " ++ display ++ " Werte", display ++ " values") -- no-spell-check
+      where
+        display = show diff
 
 partialGrade :: OutputMonad m =>  DecideInst -> [Int] -> LangM m
 partialGrade DecideInst{..} sol = do
-  preventWithHint (solLen > acLen)
+  preventWithHint (null sol)
+    (translate $ do
+      german "Lösung enthält Indizes?"
+      english "Solution contains indices?"
+    )
+    (translate $ do
+      german "Die Lösung muss mindestens einen Index enthalten."
+      english "The solution must contain at least one index."
+    )
+
+  preventIfSolutionExceedsTableSize (length sol) (getTable cnf)
+
+  pure ()
+
+
+completeGrade :: OutputMonad m => DecideInst -> [Int] -> LangM m
+completeGrade DecideInst{..} sol = (if incorrect then refuse else id) $ do
+  printWithHint (solLen > acLen)
     (translate $ do
       german "Lösung enthält nicht zu viele Indizes?"
       english "Solution does not contain too many indices?"
     )
     (translate $ do
-      german $ "Lösung enthält zu viele Indizes. Es " ++ ger ++" entfernt werden."
-      english $ "Solution contains too many indices. Please remove " ++ eng ++ " to proceed."
+      german "Lösung enthält zu viele Indizes."
+      english "Solution contains too many indices."
     )
 
-  preventWithHint (acLen > solLen)
+  printWithHint (acLen > solLen)
     (translate $ do
       german "Lösung enthält genügend Indizes?"
```

Comment:
I find it a bit weird that the output can now be something like this:
```
Solution does not contain too many indices?
No.
Solution contains too many indices.
Solution contains enough indices?
Yes.
```
If the solution contains too many indices, printing the question whether it contains enough indices is a distraction in the feedback. Maybe wrap the `printWithHint (acLen > solLen)` in a `when (acLen >= solLen)`?